test/python/preprocessing/test_csv_preprocessor.py

```diff
@@ -1,45 +1,434 @@
 import os
 import shutil
 import unittest
+import subprocess
 from pathlib import Path
-
+import random
+import numpy as np
 from marius.tools.csv_converter import general_parser
 
 TEST_DIR = "./output_dir"
+test_data_dir = "./test/test_data/"
+input_dir = "./test/test_data/test_data_copy/"
+train_file = "train_edges.txt"
+valid_file = "valid_edges.txt"
+test_file = "test_edges.txt"
+output_dir = TEST_DIR
 
 
 class TestGeneralParser(unittest.TestCase):
     """
     Tests for the general preprocessor
     """
+    @classmethod
+    def setUp(self):
+        if not Path(TEST_DIR).exists():
+            Path(TEST_DIR).mkdir()
+
+        if Path(input_dir).exists():
+            shutil.rmtree(Path(input_dir))
+
+        Path(input_dir).mkdir()
+        shutil.copy(str(Path(test_data_dir) / Path(train_file)),
+                    str(Path(input_dir) / Path(train_file)))
+        shutil.copy(str(Path(test_data_dir) / Path(valid_file)),
+                    str(Path(input_dir) / Path(valid_file)))
+        shutil.copy(str(Path(test_data_dir) / Path(test_file)),
+                    str(Path(input_dir) / Path(test_file)))
+
 
     @classmethod
     def tearDown(self):
         if Path(TEST_DIR).exists():
             shutil.rmtree(Path(TEST_DIR))
 
+        if Path(input_dir).exists():
+            shutil.rmtree(Path(input_dir))
+
+    def dataset_generator(self, train_len=1000, valid_len=100, test_len=100,
```

Comment:
This looks like it would be useful for other tests in the future. Can you create a helpers.py file in `test/python/` and move this function to there?